doc/source/conf.py

```diff
@@ -92,6 +94,14 @@
 
 myst_heading_anchors = 3
 
+# Make broken internal references into build time errors.
+# See https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-nitpicky
+# for more information. :py:class: references are ignored due to false positives
+# arising from type annotations. See https://github.com/ray-project/ray/pull/46103
+# for additional context.
+nitpicky = True
+nitpick_ignore_regex = [("py:class", ".*")]
```

Comment:
We get a bunch of false error messages from trouble that Sphinx has with type generics, which are used as annotations in various places in the code base. There's definitely other ways to deal with this issue, but this is the most straightforward right now.

In the future it might be worth investigating ways to remove this rule, because we'll be better protected against invalid class references, but for now I think this is a good start.